apisix/control/router.lua

```diff
@@ -46,6 +47,80 @@ local function format_dismod_uri(mod_name, uri)
     return core.table.concat(tmp, "")
 end
 
+local function plugins_eq(old, new)
+    local old_set = {}
+    for _, p in ipairs(old) do
+        old_set[p.name] = p
+    end
+
+    local new_set = {}
+    for _, p in ipairs(new) do
+        new_set[p.name] = p
+    end
+
+    return core.table.set_eq(old_set, new_set)
+end
+
+local function sync_local_conf_to_etcd(reset)
```

Comment:
The code is the same as it is under admin API. Why not put it into the utils directory?